### PR TITLE
add CTCSS support

### PR DESCRIPTION
**Pull request by charlie-foxtrot**
- Add support for CTCSS with new config parameter
- Remove errors for configs using "disable in channel list"
- cleanup some tabs vs spaces

### CODE REVIEW

**Comment by charlie-foxtrot** [comment]
example of the same recording handled multiple times:

- no notch or ctcss showing tone at 162.2 Hz
- notch and ctcss at the correct value of 162.2 Hz
- notch and ctcss at the wrong value of 167.9 Hz (next "standard" tone) showing the "fast" detector being wrong but the "slow" detector being correct
- notch and ctcss at the wrong value of 192.8 Hz showing both the "fast" and "slow" detectors being correct.

Screen Shot 2022-11-18 at 11 34 25 AM

File: src/squelch.cpp [hunk 1]
```diff
@@ -84,12 +84,35 @@ void Squelch::set_squelch_snr_threshold(const float &db) {
...
+	}
+	
+	if (ctcss_fast_.is_enabled()) {
+		if (ctcss_slow_.enough_samples()){
+			return ctcss_slow_.has_tone();
+		}
+		return ctcss_fast_.has_tone();
+	}
+	
+	return true;
```
Comment (charlie-foxtrot) [inline src/squelch.cpp:107]: This change introduces a bug in scanning mode.  Without this change a state of `OPENING` results in `is_open()` returning `false`, with this change it returns `true`.  The impact is a scanning SDR will be stuck on

**Comment by feelingwalnut** [comment]
this wont pick up dcs codes, correct? only ctcss.
how much work would be required to do say:
"is there a tone below 300hz Y or N, if Y open squelch if N ignore"

both ctcss and dcs exist in the subaudible range below 300hz, full ctcss and dcs support requires identifying there is a tone be it analog or digital then decoding it, then deciding to open the squelch or not based on whether it matches the code specified in settings.

while there is benefit in crowded areas to only open on specified tones, disciminating against repeater identifier pings, noise and pirate transmissions can be achieved with a simple "is a tone present or not"

**Comment by charlie-foxtrot** [comment]
> this wont pick up dcs codes, correct? only ctcss.

correct, only CTCSS

> how much work would be required to do say: "is there activity below 300hz Y or N, if Y open squelch if N ignore"
>
> both ctcss and dcs exist in the subaudible range below 300hz, full ctcss and dcs support requires identifying there is a tone be it analog or digital then decoding it, then deciding to open the squelch or not based on whether it matches the code specified in settings.
> 
> while there is benefit in crowded areas to only open on specified tones, disciminating against repeater identifier pings, noise and pirate transmissions can be achieved with a simple "is a tone present or not"

The approach in this PR (I will re-open it after some bug fixes) would not work well for "anything under 300Hz" and I am running in an environment that is re-using the same frequency for 3 transmitters with different CTCSS and I need to pick out transmissions from only one.

**Comment by feelingwalnut** [comment]
ok. thats what i was thinking. i'd reviewed the code and was pretty sure it would only work with ctcss.

one frequency i operate on sends an identifier from the repeater roughly every half hour, it drifts by a minute or so per broadcast which is sent without any coding. but while we operate with ctcss on some freqyencies, this one for some reason is dcs.

**Pull request closed without merging**